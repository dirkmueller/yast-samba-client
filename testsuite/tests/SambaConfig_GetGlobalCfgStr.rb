# encoding: utf-8

# test SambaConfig_Global::GetGlobalCfgStr:
module Yast
  class Samba_GlobalClient < Client
    def main
      Yast.import "Testsuite"
      Yast.import "SambaConfig"

      @i = {
        "global"         => {
          "a"   => 1,
          "b" => true,
          "c"=> "string",
        },
        "share" => { "sharekey" => "sharevalue" },
      }

      Testsuite.Dump(@i)

      SambaConfig.Import(@i)
      # test output of normal globals read in
      Testsuite.Dump(SambaConfig.GetGlobalCfgStr())
      j = { "a"   => "2" }
      # test override of a param
      Testsuite.Dump(SambaConfig.GetGlobalCfgStr(j))
      # test override (removal) of a param
      j["a"] = nil
      Testsuite.Dump(SambaConfig.GetGlobalCfgStr(j))

      nil
    end
  end
end

Yast::Samba_GlobalClient.new.main
