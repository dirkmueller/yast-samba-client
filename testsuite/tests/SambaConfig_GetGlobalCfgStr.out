Dump	$["global":$["a":1, "b":true, "c":"string"], "share":$["sharekey":"sharevalue"]]
Dump	[global]
Dump		a = 1
Dump		b = 1
Dump		c = string
Dump	
Dump	[global]
Dump		a = 2
Dump		b = 1
Dump		c = string
Dump	
Dump	[global]
Dump		b = 1
Dump		c = string
Dump	
